src/services/AuthService.ts

```diff
@@ -4,37 +4,31 @@ import { type NavigateFunction } from "react-router-dom"
 import { Interfaces, auth } from "../firebase"
 import { FirebaseApi } from "../firebase/api"
 import { HTTPStatusCode } from "../firebase/error"
+import { UserStore } from "../stores/UserStore"
 
 const sleep = () => new Promise((resolve) => setTimeout(resolve, 1000))
 
 class _AuthService {
-    isAdminAuth = false
-    gettingError: Error | undefined
     isGettingAuth = false
     isGettingAuthData = false
     authenticatedUser: Interfaces.IUser | undefined // Added user property to store user information
+    userStore: UserStore = new UserStore()
     private readonly auth = auth
 
     constructor() {
-        makeAutoObservable(this, {}, { deep: true })
+        makeAutoObservable(this, {})
         this.isGettingUserAuth()
     }
 
     async isGettingUserAuth(): Promise<void> {
         this.isGettingAuthData = true
         await sleep()
         const storedUser = localStorage.getItem("user")
-        // Retrieve user information from local storage
-        const storedAdminUser = localStorage.getItem("isAdmin")
-
         if (storedUser) {
             this.authenticatedUser = JSON.parse(storedUser)
+        } else {
```

Comment:
to lahko skrajšaš v eno vrstico:

`this.authenticatedUser = storedUser ? JSON.parse(storedUser) : undefined`